common/src/main/java/com/wynntils/screens/settings/widgets/FeatureSettingWidget.java

```diff
@@ -0,0 +1,308 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.screens.settings.widgets;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.config.ConfigHolder;
+import com.wynntils.core.features.Feature;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.mc.objects.CommonColors;
+import com.wynntils.mc.objects.CustomColor;
+import com.wynntils.mc.render.FontRenderer;
+import com.wynntils.mc.render.RenderUtils;
+import com.wynntils.mc.render.Texture;
+import com.wynntils.screens.settings.WynntilsSettingsScreen;
+import com.wynntils.screens.settings.elements.BooleanConfigOptionElement;
+import com.wynntils.screens.settings.elements.ConfigOptionElement;
+import com.wynntils.screens.settings.elements.DummyConfigOptionElement;
+import com.wynntils.utils.MathUtils;
+import java.util.ArrayList;
+import java.util.List;
+import net.minecraft.client.gui.components.AbstractWidget;
+import net.minecraft.client.gui.narration.NarrationElementOutput;
+import net.minecraft.network.chat.TextComponent;
+
+public final class FeatureSettingWidget extends AbstractWidget {
+    private static final int MAX_RENDER_COUNT = 6;
+
+    private static final CustomColor BORDER_COLOR = new CustomColor(86, 75, 61, 255);
+    private static final CustomColor FOREGROUND_COLOR = new CustomColor(177, 152, 120, 255);
+    private static final CustomColor SCROLLBAR_COLOR = new CustomColor(137, 117, 92, 255);
+
+    private final WynntilsSettingsScreen settingsScreen;
+
+    private Feature cachedFeature = null;
+    private List<ConfigOptionElement> configWidgets = new ArrayList<>();
+    private ConfigOptionElement hoveredConfigWidget = null;
+    private boolean enabledStateChangeable = false;
```

Comment:
This variable is unused as of now

Reply:
It is not? I check it when rendering the on off switch?

Reply:
I will get rid of it anyways if I move the switches.

Reply:
You opted to use
```java
         final Texture switchTexture = selectedFeature.isEnabled() ? Texture.SWITCH_ON : Texture.SWITCH_OFF;
```
there instead